presto-hive/src/main/java/com/facebook/presto/hive/TimestampRewriter.java

```diff
@@ -123,4 +130,30 @@ private Block modifyTimestampsInTimestampBlock(Block block, LongUnaryOperator mo
 
         return blockBuilder.build();
     }
+
+    private Block modifyTimestampsInArrayBlock(Type type, Block block, LongUnaryOperator modification)
+    {
+        if (block instanceof AbstractArrayBlock) {
+            AbstractArrayBlock arrayBlock = (AbstractArrayBlock) block;
+            Block innerBlock = modifyTimestampsInBlock(arrayBlock.getValues(),
+                    type.getTypeParameters().get(0),
+                    modification);
+
+            return new ArrayBlock(arrayBlock.getPositionCount(), arrayBlock.getValueIsNull(), arrayBlock.getOffsets(), innerBlock);
+        }
+        else {
+            // Dear reviewer: This is slow path for sake when ARRAY is not represented as either AbstractArrayBlock or Lazy block wrapping AbstractArrayBlock.
+            //                Do we need this? Or should we just throw IllegalArgumentException("unsupported block type") here?
```

Comment:
maybe add logging here, so we could have a knowledge how often this path is used?